crates/subspace-networking/src/behavior/persistent_parameters.rs

```diff
@@ -251,6 +232,28 @@ impl NetworkingParametersRegistry for NetworkingParametersManager {
         self.cache_need_saving = true;
     }
 
+    async fn remove_known_peer_addresses(&mut self, peer_id: PeerId, addresses: Vec<Multiaddr>) {
+        addresses
+            .into_iter()
+            .map(|addr| {
+                // remove p2p-protocol suffix if any
+                let mut modified_address = addr.clone();
+
+                if let Some(Protocol::P2p(_)) = modified_address.pop() {
+                    modified_address
+                } else {
+                    addr
+                }
+            })
+            .for_each(|addr| {
+                if let Some(addresses) = self.known_peers.get_mut(&peer_id) {
```

Comment:
`.get_mut` will update recency of the record in LRU, are you sure you didn't want to use `.peek_mut` instead?